keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1237 @@
+
+      4. [Already exists] Sum of container level requests less than container level limits is invalid.
+
+
+#### Scheduler Changes
+
+The scheduler logic needs to be updated to support pod-level requests and limits.
+
+* **Scenario 1: Only container-level resources are specified**
+
+  No changes, same as existing.
+
+* **Scenario 2: Only pod-level resources are specified**
+
+  Consider pod-level requests to find the suitable node for pod placement.
+
+* **Scenario 3: Both container-level and pod-level resources are specified**
+
+  Scheduler will prioritize pod-level requests as the sole indicator of the pod’s resources needs to find a node that will fulfill the overall 
+  resource requirements of the pod. It will ignore container level requests.
+
+#### QoS Changes
+
+The QoS definitions need to be updated to consider pod-level resources. To attach a QoS class to a pod, the  pod’s resource specifications are evaluated in order:
+
+1. The values for resources specified in the pod’s `resources` stanza.
+2. The value of resources specified in the container’s `resources` stanza.
+
+In simple words, if `resources` is specified at pod-level, the values in pod-level `resources` stanza are given a preference in determining the QoS class and container
+level `resources` stanza(s) is ignored
+
+* **Guaranteed QoS Criteria**
+
+    A pod can qualify for Guaranteed QoS under two distinct scenarios:
+
+    1. Explicit Pod-level Resources
+
+       * Pod level requests and limits are specified for each resource.
+       * Requests and limits are equal or each resource. 
+      
+       In this case, container level resource specifications, specified or not specified, are irrelvant to QoS determination.
+
+    2. Implicit Pod-level Resources (via containers)
+       * Pod level `resources` setting is not specified.
+       * *Every* container within the pod has requests and limits specified for each resource.
+       * All container resource requests and limits are equal.
+
+  **Note: If the pod spec includes `resources` stanza at pod level, but either requessts or limits are missing for a resource, the pod will not be considered
+  Guaranteed QoS, even if container-level requests and limits are present and are equal.**
+
+* **Burstable QoS Criteria**
+
+    A pod can qualify for Burstable QoS in one of the following scenarios:
+
+    1. Explicit Pod-level Resources
+
+       * It doesn't meet the criteria of Guaranteed QoS. 
+       * At least one of pod-level requests or limits is specified for CPU or memory.
+
+    2. Implicit Pod-level Resources (via containers)
+
+       * It doesn't meet the criteria of Guaranteed QoS. 
+       * Pod level `resources` setting is not specfied.
+       * At least one Container in the Pod has a memory or CPU request or limit.
+
+* **BestEffort QoS Criteria**
+
+    None of pod-level or container-level resources requests or limits are specified.
+
+#### Init Containers & Sidecar Containers
```

Comment:
You should put this section earlier, and then define sceduler and QoS behavior in terms of it, I think?